transporters/src/main/java/com/nextdoor/bender/ipc/tcp/TcpTransportBuffer.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * Copyright 2017 Nextdoor.com, Inc
+ *
+ */
+
+package com.nextdoor.bender.ipc.tcp;
+
+import com.nextdoor.bender.InternalEvent;
+import com.nextdoor.bender.ipc.TransportBuffer;
+import com.nextdoor.bender.ipc.TransportSerializer;
+import java.io.Closeable;
+import okio.Buffer;
+
+public class TcpTransportBuffer implements TransportBuffer, Closeable {
+
+  private final TransportSerializer serializer;
+  private final Buffer buffer;
+  private final long maxBufferSize;
+
+  public TcpTransportBuffer(TcpTransportConfig config, TransportSerializer serializer) {
+    this.serializer = serializer;
+    this.buffer = new Buffer();
+    this.maxBufferSize = config.getMaxBufferSize();
+  }
+
+  @Override
+  public boolean add(InternalEvent ievent) throws IllegalStateException {
+    byte[] data = serializer.serialize(ievent);
+    if (buffer.size() + data.length >= maxBufferSize) {
```

Comment:
I'm hitting `com.nextdoor.bender.ipc.TransportException: Buffer is full despite being newly created` https://github.com/Nextdoor/bender/blob/master/core/src/main/java/com/nextdoor/bender/ipc/IpcSenderService.java#L113 when I set the lambda kinesis batch size to 10,000 events.

Reply:
Thanks for the feedback @stlava !
I can't reproduce this problem, do you have more details to share ?
Do you still reproduce it with the changes I pushed ?

Reply:
Tested it out again this morning. It's only selectively happening but more frequently as I increase the kinesis batch size.

Reply:
Ok the issue was on our side. A service was writing logs greater than the max buffer size and increasing the buffer fixed it. However, I do think the max on the transport should be larger to accommodate a larger range of configurations. Maybe 10mb?

Reply:
Sure !